waiter/src/waiter/auth/authentication.clj

```diff
@@ -44,12 +48,13 @@
 (defn handle-request-auth
   "Invokes the given request-handler on the given request, adding the necessary
   auth headers on the way in, and the x-waiter-auth cookie on the way out."
-  [handler request user principal password]
-  (let [auth-params-map (auth-params-map principal user)
-        handler' (middleware/wrap-merge handler auth-params-map)]
-    (-> request
-        handler'
-        (add-cached-auth password principal))))
+  ([handler request principal password]
+   (handle-request-auth handler request principal (auth-params-map principal) password))
+  ([handler request principal auth-params-map password & [age-in-seconds]]
```

Comment:
Same overloading change request especially since this function is already overloaded.
https://stuartsierra.com/2015/06/01/clojure-donts-optional-arguments-with-varargs